src/lib/utils/meta_tables/meta_log_table.cpp

```diff
@@ -0,0 +1,37 @@
+#include "meta_log_table.hpp"
+
+#include "constant_mappings.hpp"
+#include "hyrise.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+MetaLogTable::MetaLogTable()
+    : AbstractMetaTable(TableColumnDefinitions{{"timestamp", DataType::String, false},
+                                               {"log_level", DataType::String, false},
+                                               {"log_level_id", DataType::Int, false},
+                                               {"reporter", DataType::String, false},
+                                               {"message", DataType::String, false}}) {}
+
+const std::string& MetaLogTable::name() const {
+  static const auto name = std::string{"log"};
+  return name;
+}
+
+std::shared_ptr<Table> MetaLogTable::_on_generate() const {
+  auto output_table = std::make_shared<Table>(_column_definitions, TableType::Data, std::nullopt, UseMvcc::Yes);
+
+  for (const auto& entry : Hyrise::get().log_manager.log_entries()) {
+    std::ostringstream timestamp;
+    auto time = std::chrono::system_clock::to_time_t(entry.timestamp);
+    struct tm buffer;
+    timestamp << std::put_time(localtime_r(&time, &buffer), "%F %T");
```

Comment:
Oh god. Maybe link to https://stackoverflow.com/questions/25618702/why-is-there-no-c11-threadsafe-alternative-to-stdlocaltime-and-stdgmtime so that people know that this is ugly because of external constraints.

Reply:
(Tidy failed around here, too)